pkg/testmachinery/testflow/validation.go

```diff
@@ -99,7 +107,45 @@ func Validate(identifier string, tf tmv1beta1.TestFlow, locs locations.Locations
 	return nil
 }
 
-func checkIfDependentStepsExist(identifier string, tf tmv1beta1.TestFlow, usedStepNames map[string]bool) error {
+func ensureArtifactsFromStepsExist(identifier string, tf tmv1beta1.TestFlow, usedStepNames map[string]*tmv1beta1.DAGStep) error {
+	for stepName, step := range usedStepNames {
```

Comment:
The identifier in line 112 is only used for the user to print out errors and show exactly where the error occurs.
But here the user doenst know anything about our internal list.

Therefor I would loop through the the testflow list so we can point to the exact erroneous step.
```suggestion
	for i, step := range tf {
```

Reply:
done